File: Math/Easy/258/258.py
# class Solution:
#     def addDigits(self, num: int) -> int:
#         sum = 0
#         while num:
#             sum += num % 10
#             num //= 10
#         if sum < 10:
#             return sum
#         return self.addDigits(sum)
    
class Solution:
    def addDigits(self, num: int) -> int:
        while num > 9:
            sum = 0
            while num:
                sum += num % 10
                num //= 10
            num = sum
        return num

File: Math/Easy/258/258.c
// int addDigits(int num){
//     while(num > 9){
//         int sum = 0;
//         while(num){
//             sum += num % 10;
//             num /= 10;
//         }
//         num = sum;
//     }
//     return num;
// }

int addDigits(int num){
    int sum = 0;
    while(num){
        sum += num % 10;
        num /= 10;
    }
    if(sum < 10){
        return sum;
    }
    return addDigits(sum);
}
